src/Apps/Order2/Routes/Checkout/Hooks/useStripePaymentBySetupIntentId.tsx

```diff
@@ -38,69 +29,65 @@ export function useStripePaymentBySetupIntentId(
   )
 
   useEffect(() => {
-    // pull necessary params from Stripe redirect URL
-    const urlParams = new URLSearchParams(window.location.search)
-    const setup_intent = urlParams.get("setup_intent")
-    const redirect_status = urlParams.get("redirect_status")
-    const save_bank_account = urlParams.get("save_bank_account")
-    const confirmation_token = urlParams.get("confirmation_token")
-
-    // Convert string to boolean - URLSearchParams returns string "false" or "true"
-    const shouldSaveBankAccount = save_bank_account === "true"
-    const oneTimeUse = !shouldSaveBankAccount
+    const handleRedirect = async () => {
+      // pull necessary params from Stripe redirect URL
+      const urlParams = new URLSearchParams(window.location.search)
+      const setup_intent = urlParams.get("setup_intent")
+      const redirect_status = urlParams.get("redirect_status")
+      const save_bank_account = urlParams.get("save_bank_account")
+      const confirmation_token = urlParams.get("confirmation_token")
 
-    if (setup_intent && redirect_status === "succeeded") {
-      setIsProcessingRedirect(true)
-
-      const isOffer = orderData.mode === "OFFER"
-      if (isOffer) {
-        setOfferAmountComplete()
+      // If no Stripe redirect params, call onComplete immediately
+      if (!setup_intent || redirect_status !== "succeeded") {
+        onComplete?.()
+        return
       }
```

Comment:
not sure this is right. if there is a non-success status then it might be an error we need to handle.

the onComplete callback arg just says that this processing is complete for the loading state logic to proceed.

Reply:
leaving it as-is for now because we have other work coming that will look at the mechanics more closely.